### PR TITLE
[CAS-1252] Fix WS issues

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-1252

### 🎯 Goal

_Describe why we are making this change_

### 🛠 Implementation details

_Provide a description of the implementation_

### 🎨 UI Changes

_ Add relevant screenshots_

| Before | After |
| --- | --- |
| img | img |

### 🧪 Testing

_Explain how this change can be tested (or why it can't be tested)_

### ☑️ Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [ ] Reviewers added

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt [hunk 1]
```diff
@@ -55,6 +55,10 @@ internal class NetworkStateProvider(private val connectivityManager: Connectivit
     }
 
     fun subscribe(listener: NetworkStateListener) {
+        if (listeners.contains(listener)) {
+            return
+        }
+
```
Comment (bychkovdmitry) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt:61]: Prevent adding more listeners on network availability change

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt [hunk 2]
```diff
@@ -55,6 +55,10 @@ internal class NetworkStateProvider(private val connectivityManager: Connectivit
     }
 
     fun subscribe(listener: NetworkStateListener) {
+        if (listeners.contains(listener)) {
```
Comment (ogkuzmin) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt:58]: You can change the type of this container to `Set` and avoid manual check

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt [hunk 3]
```diff
@@ -86,6 +86,7 @@ internal class ChatSocketServiceImpl constructor(
                         callListeners { it.onDisconnected(DisconnectCause.NetworkNotAvailable) }
                     }
                     is State.DisconnectedByRequest -> {
+                        networkStateProvider.unsubscribe(networkStateListener)
```
Comment (bychkovdmitry) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/socket/ChatSocketServiceImpl.kt:89]: Stop listening to network changes when app is in the background

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt (hunk 2 above)
Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/network/NetworkStateProvider.kt:58]: This is just a raw PR to showcase the issue. Also `listeners + listener` internally creates a new collection.

**Comment by JcMinarro** [comment]
Closing it.
It has been fixed with other improvement on #2269

**Pull request closed without merging**